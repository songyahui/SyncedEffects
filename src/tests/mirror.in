;
;
I3(lala);
I4(34) I1 I3(hihihiihi)
;
I4(5);
I4(5);
