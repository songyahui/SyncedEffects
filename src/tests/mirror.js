"use hopscript"

var hh = require("hiphop");

function foo(evt) {
   console.log("hi from foo signal", evt.type, "is set!")
}

function bar(evt) {
   console.log("hi from bar signal", evt.type, "is set with value",
	       evt.nowval, "!")
}

var inSig={accessibility:hh.IN};

var prg = <hh.module I1=${inSig} I2=${inSig} I3=${inSig} I4=${inSig} O1 O2 O3 O4>
  <hh.loop>
    <hh.sequence>
      <hh.if I1>
	<hh.emit O1/>
      </hh.if>
      <hh.if I2>
	<hh.emit O2 apply=${function() {return this.I2.nowval}}/>
      </hh.if>
      <hh.if I3>
	<hh.emit O3 apply=${function() {return this.I3.nowval}}/>
      </hh.if>
      <hh.if I4>
	<hh.emit O4 apply=${function() {return this.I4.nowval}}/>
      </hh.if>
      <hh.pause/>
    </hh.sequence>
  </hh.loop>
</hh.module>

var m = new hh.ReactiveMachine(prg, "mirror");
m.addEventListener("O1", foo);
m.addEventListener("O2", bar);
m.addEventListener("O3", bar);
m.addEventListener("O4", bar);

exports.prg = m
