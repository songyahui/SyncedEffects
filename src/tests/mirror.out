mirror> ;
--- Output:
mirror> ;
--- Output:
mirror> I3(lala);
hi from bar signal O3 is set with value lala !
--- Output: O3("lala")
mirror> I4(34) I1 I3(hihihiihi);
hi from foo signal O1 is set!
hi from bar signal O3 is set with value hihihiihi !
hi from bar signal O4 is set with value 34 !
--- Output: O1 O3("hihihiihi") O4(34)
mirror> I4(5);
hi from bar signal O4 is set with value 5 !
--- Output: O4(5)
mirror> I4(5);
hi from bar signal O4 is set with value 5 !
--- Output: O4(5)
